src/pal/tools/gen-buildsys-clang.sh

```diff
@@ -168,17 +158,18 @@ if [[ "$scan_build" == "ON" ]]; then
     cmake_command="$SCAN_BUILD_COMMAND $cmake_command"
 fi
 
+# Include CMAKE_USER_MAKE_RULES_OVERRIDE as uninitialized since it will hold its value in the CMake cache otherwise can cause issues when branch switching
 $cmake_command \
   -G "$generator" \
-  "-DCMAKE_USER_MAKE_RULES_OVERRIDE=${__currentScriptDir}/$overridefile" \
   "-DCMAKE_AR=$llvm_ar" \
   "-DCMAKE_LINKER=$llvm_link" \
   "-DCMAKE_NM=$llvm_nm" \
   "-DCMAKE_OBJDUMP=$llvm_objdump" \
   "-DCMAKE_BUILD_TYPE=$buildtype" \
-  "-DCMAKE_EXPORT_COMPILE_COMMANDS=1 " \
```

Comment:
Why was this removed?

Reply:
The goal here was to centralize where we set `CMAKE_EXPORT_COMPILE_COMMANDS` to a single location in the tree (as part of limiting the number of defines we have to pass to CMake on the command line). I put it in the root CMakeLists.txt file. I'll can move it to configurecompiler.cmake if we want the compilation database generated for the test build as well.